Change password input for gpg from STDIN to dedicated fd

Narrat opened the pull request:
As a draft  because of the same reasons mentioned in #585 
___________________
Background of this PR is the issue that tomb has issues with passwords that try to use characters in a combination so they create control sequences for the shell. For example newline `\n`, form feed `\f` and tabulator `\t`.
This is due how the password is provided for the gpg call. gpg is used with `--passphrase-fd 0` which reads the password from `STDIN` until it encounters a `newline`. Unfortunately this means the shell interprets those character sequences in the password itself. Especially fatal in case of \n as this will reduce the password to this point. Example: you want to set password `test\ntest`. gpg, while reading `STDIN`, will stop at `\n` and the resulting password will be only `test`.
Not sure yet what happens with the rest, but it does seem to be discarded in general and not added to or used as `TOMBSECRET`.

Two ways to avoid interpreting control sequences:

* change `--passphrase-fd` to a dedicated descriptor above 2 (like `--passphrase-fd 3` and input password like `3<<<"$password"`
 * use `--passphrase-file` and use an anonymous pipe

In general the first option is similar to the current solution, just changing it to a different FD, which allows to avoid interpreting control sequences. But is untested if the redirect isn't visible while tracing.
This solutions changes the fd for the password input from STDIN to a new one file-descriptor (3 in this example).

Narrat committed:
gpg_decrypt: drop GnuPG 1.4.11 workaround

Remove last remnant of 9e820f3de6a1b3ecab0662d0ae30c0b79d6d4e43
The latest release of gpg1 is 1.4.23 from 2018. It is safe to assume, that there
is no modern distribution, which ships this specific gpg1 version.
As the workaround itself only applies to 1.4.11, any version before or after seem
to be safe.
This removal doesn't remove support for gpg1 in general as it isn't end-of-life.
All the options in use are available with gpg1 and only a broken --status-fd
seemed to be of concern.

Closes #581

jaromil commented:
I'm in favor of this, yet we'll need to state the introduction of a breaking change for anyone who uses passwords with control characters which were previously misinterpreted.

Narrat committed:
Change password input for gpg operations

GPG was used in way, that the password for the payload and the payload
was piped into GPG separated by a newline ("--passphrase-fd 0" where 0
is STDIN). It will then read the input until a newline is encountered
and strip it from the pipe. After which only the encrypted payload
remains.
This has some consequences for passwords in general, as the running
shell is involved and may interpret ANSI Escape Sequences.

ANSI Escape sequences can be used for various tasks and are
non-printable characters.
Build with a backslash ("\"), they create control sequences
like \n (newline), \t (horizontal tab) and \f (formfeed).
If those are used in passwords various stuff can happen.
\n for example will shorten the password depending on its position.
\ at the end of password will nullify the newline separator between the
password and the encrypted payload.
See KNOWN_BUGS.md for a table with some cases.

To solve those unintended changes the file descriptor is changed from 0
(STDIN) to 3. Which is the next free one after 1 (STDOUT) and 2 (STDERR).
The password will be redirected via "3<<<$password".

This solution won against a switch to --passphrase-file and the use of an
anonymous pipe <(print -R -n - "$password").
In general the use of --passphrase-file is discouraged, but that should
only apply if a real file is used. With the pipe only the respective file
descriptor should be seen and not its content. For which only gpg is
allowed to have access. And if gpg is completed the pipe and its content
is gone.

What changes:
pinentry or more accuratly assuan will still output a (percent)formated
password, but the shell won't interfere anymore.

This does not address passwords that contain % and the difference
between entering the password via pinentry and --tomb-pwd.

Addresses #304
